kvdb/postgres/db.go

```diff
@@ -0,0 +1,174 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/btcsuite/btcwallet/walletdb"
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+const (
+	hexTableId      = "hex"
+	sequenceTableId = "sequence"
+
+	systemTablePrefixExtension = "sys"
+)
+
+// KV stores a key/value pair.
+type KV struct {
+	key string
+	val string
+}
+
+// db holds a reference to the postgres connection connection.
+type db struct {
+	dsn string
+
+	// Table name prefix to simulate namespaces. We don't use schemas
+	// because at least sqlite does not support that.
```

Comment:
It isn't that simple, because SQL isn't a universal language. Some of the queries need to be rewritten for sqlite. But other than that, there isn't much that changes.